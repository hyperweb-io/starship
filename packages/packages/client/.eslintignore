node_modules/
dist/
main/
module/
coverage/
prompt.js
