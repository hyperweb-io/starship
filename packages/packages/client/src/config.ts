export interface Ports {
  rest?: number;
  rpc?: number;
  grpc?: number;
  'grpc-web'?: number;
  exposer?: number;
  faucet?: number;
  prometheus?: number;
  grafana?: number;
  cometmock?: number;
  ws?: number;
}

export interface Resources {
  cpu: string | number;
  memory: string | number;
  limits?: {
    cpu: string | number;
    memory: string | number;
  };
  requests?: {
    cpu: string | number;
    memory: string | number;
  };
}

export interface FaucetConfig {
  enabled: boolean;
  type?: string;
  image?: string;
  concurrency?: number;
  ports?: Ports;
  resources?: Resources;
}

export interface Chain {
  id: string | number;
  name: string;
  numValidators: number;
  image?: string;
  home?: string;
  binary?: string;
  prefix?: string;
  denom?: string;
  prettyName?: string;
  coins?: string;
  hdPath?: string;
  coinType?: number;
  metrics?: boolean;
  repo?: string;
  assets?: any[]; // Define more specifically if asset structure is known
  upgrade?: {
    enabled: boolean;
    type?: string;
    genesis?: string;
    upgrades?: {
      name: string;
      version: string;
    }[];
  };
  faucet?: FaucetConfig;
  ports?: Ports;
  scripts?: {
    createGenesis?: Script;
    updateGenesis?: Script;
    updateConfig?: Script;
    createValidator?: Script;
    transferTokens?: Script;
    buildChain?: Script;
    ibcConnection?: Script;
  };
  ics?: {
    enabled: boolean;
    image?: string;
  };
  cometmock?: {
    enabled: boolean;
    image?: string;
  };
  resources?: Resources;
}

export interface Script {
  file?: string;
  data?: string;
}

export interface Relayer {
  name: string;
  type: string;
  image?: string;
  replicas?: number;
  chains: string[];
  config?: any; // Define more specifically if structure is known
  channels?: {
    aChain: string;
    bChain: string;
    aPort: string;
    bPort: string;
    aConnection: string;
    newConnection?: boolean;
    channelVersion?: string;
    order?: string;
  }[];
  resources?: Resources;
  ports?: Ports;
}

export interface Explorer {
  enabled: boolean;
  type: string;
  image?: string;
  localhost?: boolean;
  ports?: Ports;
  resources?: Resources;
}

export interface Registry {
  enabled: boolean;
  image: string;
  localhost?: boolean;
  ports?: Ports;
  resources?: Resources;
}

export interface Monitoring {
  enabled: boolean;
  ports?: {
    prometheus?: number;
    grafana?: number;
  };
  resources?: Resources;
}

export interface Ingress {
  enabled: boolean;
  type: string;
  host?: string;
  certManager?: {
    issuer?: string;
  };
  resources?: Resources;
}

export interface Images {
  imagePullPolicy: string;
}

export interface Frontend {
  name: string;
  type: string;
  image: string;
  replicas?: number;
  ports?: Ports;
  env?: Record<string, string>;
  resources?: Resources;
}

export interface StarshipConfig {
  name: string;
  version: string;
  chains: Chain[];
  relayers?: Relayer[];
  explorer?: Explorer;
  registry?: Registry;
  monitoring?: Monitoring;
  ingress?: Ingress;
  images?: Images;
  frontends?: Frontend[];
}
