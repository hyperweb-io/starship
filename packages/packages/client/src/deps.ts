import * as shell from 'shelljs';

export type Dependency = {
  name: string;
  url: string;
  macUrl?: string; // Optional property for macOS-specific URLs
  installed: boolean;
};

export const dependencies: Dependency[] = [
  {
    name: 'kubectl',
    url: 'https://kubernetes.io/docs/tasks/tools/',
    macUrl: 'https://docs.docker.com/desktop/install/mac-install/',
    installed: !!shell.which('kubectl')
  },
  {
    name: 'docker',
    url: 'https://docs.docker.com/get-docker/',
    macUrl: 'https://docs.docker.com/desktop/install/mac-install/',
    installed: !!shell.which('docker')
  },
  {
    name: 'helm',
    url: 'https://helm.sh/docs/intro/install/',
    installed: !!shell.which('helm')
  }
];
