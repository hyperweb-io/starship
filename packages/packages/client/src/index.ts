export * from './client';
export * from './installer';
