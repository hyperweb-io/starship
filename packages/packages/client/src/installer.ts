import chalk from 'chalk';
import * as os from 'os';
import * as shell from 'shelljs';

type Installation = {
  mac: string;
  linux: string;
};

export class StarshipInstaller {
  private installations: Record<string, Installation> = {
    docker: {
      mac: 'Please install Docker. Follow: https://docs.docker.com/desktop/install/mac-install/',
      linux:
        'Please install Docker. Follow: https://docs.docker.com/engine/install/ubuntu/'
    },
    kubectl: {
      mac: 'brew install kubectl',
      linux: `curl -Lks "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" > ~/.local/bin/kubectl && chmod +x ~/.local/bin/kubectl`
    },
    helm: {
      mac: 'brew install helm',
      linux:
        'curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash'
    }
  };

  async checkAndInstallBinary(binaryName: string) {
    if (!shell.which(binaryName)) {
      console.log(`${binaryName} is not installed`);
      await this.installBinary(binaryName);
      if (!shell.which(binaryName)) {
        console.log(
          chalk.red(
            `Installation of ${binaryName} failed. Please install ${binaryName} manually.`
          )
        );
        process.exit(1);
      }
    }
  }

  async checkAndInstallDependencies() {
    try {
      for (const dependency of Object.keys(this.installations)) {
        console.log(chalk.yellow(`Checking ${dependency}...`));
        await this.checkAndInstallBinary(dependency);
      }
      console.log(chalk.green('All dependencies are installed!! Good to go!!'));
    } catch (err) {
      console.error(chalk.red(`Error installing dependencies: ${err}`));
    }
  }

  async installBinary(binaryName: string) {
    const platform = os.platform();
    const installation = this.installations[binaryName];
    if (platform === 'darwin') {
      await this.runInstallation(installation.mac);
    } else if (platform === 'linux') {
      await this.runInstallation(installation.linux);
    }
  }

  private async runInstallation(command: string) {
    shell.exec(command);
  }
}
