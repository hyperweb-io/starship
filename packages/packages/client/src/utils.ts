import axios from 'axios';

export const handleAxiosError = (error: unknown): string => {
  if (axios.isAxiosError(error)) {
    // Handle connection refused errors
    if (error.code === 'ECONNREFUSED') {
      return `Connection refused to ${error.config?.url}`;
    }

    // Handle other axios errors
    if (error.response) {
      return `HTTP ${error.response.status}: ${error.response.statusText}`;
    }

    // Handle network errors
    if (error.request) {
      return `Network error: ${error.message}`;
    }

    // Handle other axios errors
    return error.message;
  }

  // Handle non-axios errors
  if (error instanceof Error) {
    return error.message;
  }

  return 'Unknown error';
};
