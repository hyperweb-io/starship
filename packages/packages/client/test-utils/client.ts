import { relative } from 'path';
import strip from 'strip-ansi';

import { StarshipClient } from '../src';
import { config } from './config';

interface TestClientCtx {
  commands: string[];
  logs: string[];
  code: number;
}

export const createClient = () => {
  const ctx: TestClientCtx = {
    commands: [],
    logs: [],
    code: -1
  };

  const client = new StarshipClient({
    name: 'osmojs',
    config: relative(process.cwd(), config.configPath)
  });

  const handler = {
    get(target: any, prop: string, _receiver: any) {
      const originalMethod = target[prop];
      if (typeof originalMethod === 'function') {
        return function (...args: any[]) {
          const argsString = args
            .map((arg) => strip(JSON.stringify(arg)))
            .join(', ');
          ctx.logs.push(`Call: ${prop}(${argsString})`);
          // if you want to see nested Call, replace target with this
          // I double checked both this and target, it does not call the exec in the methods when used internally
          return originalMethod.apply(target, args);
        };
      }
      return originalMethod;
    }
  };

  const proxiedClient: StarshipClient = new Proxy(client, handler);

  // Overriding the exit method
  // @ts-expect-error - Ignore lint error
  proxiedClient.exit = (code: number) => {
    ctx.code = code;
  };

  // @ts-expect-error - Ignore lint error
  proxiedClient.ensureFileExists = (_filename: string) => {};

  // Overriding the exec method
  // @ts-expect-error - Ignore lint error
  proxiedClient.exec = (cmd: string[]) => {
    // @ts-expect-error - Ignore lint error
    client.checkDependencies();
    ctx.commands.push(cmd.join(' '));
    ctx.logs.push(cmd.join(' '));
    return '';
  };

  // Overriding the log method
  // @ts-expect-error - Ignore lint error
  proxiedClient.log = (cmd: string) => {
    const str = strip(cmd);
    if (/\n/.test(str)) {
      ctx.logs.push('Log⬇\n' + str + '\nEndLog⬆');
    } else {
      ctx.logs.push('Log: ' + str);
    }
  };

  return {
    client: proxiedClient,
    ctx
  };
};

export const expectClient = (ctx: TestClientCtx, code: number) => {
  expect(ctx.logs.join('\n')).toMatchSnapshot();
  expect(ctx.commands.join('\n')).toMatchSnapshot();
  expect(ctx.code).toBe(code);
};
