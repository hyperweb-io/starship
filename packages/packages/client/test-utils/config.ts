import { readFileSync } from 'fs';
import * as yaml from 'js-yaml';
import { join, resolve } from 'path';

import { StarshipConfig } from '../src/config';

export const fixtureDir = resolve(join(__dirname, '/../../../__fixtures__'));
export const outputDir = resolve(join(__dirname, '/../../../__output__'));

function loadConfig(filename: string) {
  const configPath = join(fixtureDir, filename);
  const configAsYaml = readFileSync(configPath, 'utf-8');
  const config: StarshipConfig = yaml.load(configAsYaml) as StarshipConfig;
  return {
    configPath,
    configAsYaml,
    config
  };
}

const localConfig = loadConfig('local-config.yaml');
const config = loadConfig('config.yaml');

export { config, localConfig };
