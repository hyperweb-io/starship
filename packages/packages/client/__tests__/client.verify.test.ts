import axios from 'axios';

import { StarshipConfig } from '../src/config';
import { createClient, expectClient } from '../test-utils/client';
import { config } from '../test-utils/config';

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('StarshipClient verify', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should verify all services successfully', async () => {
    const { client, ctx } = createClient();
    client.dependencies.forEach((dep) => (dep.installed = true));
    client.setConfig(config.config);

    // Mock successful responses for all services
    mockedAxios.get.mockImplementation((url) => {
      // Chain REST check (port 1317, 1313)
      if (url.includes('/cosmos/bank/v1beta1/supply')) {
        return Promise.resolve({
          status: 200,
          data: {
            supply: [{ amount: '1000000' }]
          }
        });
      }

      // Chain RPC check (port 26657, 26653)
      if (
        url.includes('/status') &&
        (url.includes('26657') || url.includes('26653'))
      ) {
        return Promise.resolve({
          status: 200,
          data: {
            result: {
              sync_info: {
                latest_block_height: '100'
              }
            }
          }
        });
      }

      // Chain faucet check (port 8007, 8003)
      if (
        url.includes('/status') &&
        (url.includes('8007') || url.includes('8003'))
      ) {
        // Extract port from URL
        const port = url.includes('8007') ? 8007 : 8003;
        // Find the chain ID based on the port
        const chain = config.config.chains?.find(
          (chain) => chain.ports?.faucet === port
        );
        return Promise.resolve({
          status: 200,
          data: {
            chainId: chain?.id || 'unknown'
          }
        });
      }

      // Registry check (port 8081)
      if (url.includes('/chains')) {
        return Promise.resolve({
          status: 200,
          data: {
            chains: config.config.chains?.map((chain) => chain.id) || []
          }
        });
      }

      // Explorer check (port 8080)
      if (url.includes('8080')) {
        return Promise.resolve({
          status: 200,
          data: '<html><body>Ping Dashboard</body></html>'
        });
      }

      // Throw error for unhandled URLs
      throw new Error(`Unhandled URL in mock: ${url}`);
    });

    await client.verify();
    expectClient(ctx, 0);
  });

  it('should handle registry verification failure', async () => {
    const { client, ctx } = createClient();
    client.dependencies.forEach((dep) => (dep.installed = true));
    client.setConfig(config.config);

    // Mock registry failure
    mockedAxios.get.mockImplementation((url) => {
      if (url.includes('/chains')) {
        return Promise.reject(new Error('Registry not available'));
      }

      // Throw error for unhandled URLs
      throw new Error(`Unhandled URL in mock: ${url}`);
    });

    await client.verify();
    expectClient(ctx, 1);
  });

  it('should handle explorer verification failure', async () => {
    const { client, ctx } = createClient();
    client.dependencies.forEach((dep) => (dep.installed = true));
    client.setConfig(config.config);

    // Mock explorer failure
    mockedAxios.get.mockImplementation((url) => {
      if (url.includes('8080')) {
        return Promise.reject(new Error('Explorer not available'));
      }

      // Throw error for unhandled URLs
      throw new Error(`Unhandled URL in mock: ${url}`);
    });

    await client.verify();
    expectClient(ctx, 1);
  });

  it('should handle relayer verification failure', async () => {
    const { client, ctx } = createClient();
    client.dependencies.forEach((dep) => (dep.installed = true));
    client.setConfig(config.config);

    // Mock relayer failure
    mockedAxios.get.mockImplementation((url) => {
      if (url.includes('/version')) {
        return Promise.reject(new Error('Relayer not available'));
      }

      // Throw error for unhandled URLs
      throw new Error(`Unhandled URL in mock: ${url}`);
    });

    await client.verify();
    expectClient(ctx, 1);
  });

  it('should skip disabled services', async () => {
    const { client, ctx } = createClient();
    client.dependencies.forEach((dep) => (dep.installed = true));

    // Create config with disabled services
    const disabledConfig: StarshipConfig = {
      ...config.config,
      explorer: {
        ...config.config.explorer,
        enabled: false
      },
      registry: {
        ...config.config.registry,
        enabled: false
      }
    };
    client.setConfig(disabledConfig);

    // Mock successful responses for enabled services
    mockedAxios.get.mockImplementation((url) => {
      // Chain REST check (port 1317, 1313)
      if (url.includes('/cosmos/bank/v1beta1/supply')) {
        return Promise.resolve({
          status: 200,
          data: {
            supply: [{ amount: '1000000' }]
          }
        });
      }

      // Chain RPC check (port 26657, 26653)
      if (
        url.includes('/status') &&
        (url.includes('26657') || url.includes('26653'))
      ) {
        return Promise.resolve({
          status: 200,
          data: {
            result: {
              sync_info: {
                latest_block_height: '100'
              }
            }
          }
        });
      }

      // Chain faucet check (port 8007, 8003)
      if (
        url.includes('/status') &&
        (url.includes('8007') || url.includes('8003'))
      ) {
        // Extract port from URL
        const port = url.includes('8007') ? 8007 : 8003;
        // Find the chain ID based on the port
        const chain = disabledConfig.chains?.find(
          (chain) => chain.ports?.faucet === port
        );
        return Promise.resolve({
          status: 200,
          data: {
            chainId: chain?.id || 'unknown'
          }
        });
      }

      // Throw error for unhandled URLs
      throw new Error(`Unhandled URL in mock: ${url}`);
    });

    await client.verify();
    expectClient(ctx, 0);
  });
});
