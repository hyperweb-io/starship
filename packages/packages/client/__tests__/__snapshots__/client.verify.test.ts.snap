// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarshipClient verify should handle explorer verification failure 1`] = `
"Call: setConfig({"name":"osmojs","version":"v0.2.3","chains":[{"id":"osmosis-1","name":"osmosis","image":"pyramation/osmosis:v16.1.0","numValidators":1,"ports":{"rest":1317,"rpc":26657,"faucet":8007}},{"id":"cosmos-2","name":"cosmos","numValidators":1,"ports":{"rest":1313,"rpc":26653,"faucet":8003}}],"relayers":[{"name":"osmos-cosmos","type":"hermes","replicas":1,"chains":["osmosis-1","cosmos-2"]}],"registry":{"enabled":true,"ports":{"rest":8081,"grpc":9091}}})
Log: config again: [object Object]
Call: verify()
Log: Verifying services...
Log: FAILURE chain-osmosis-1 (rest): Unhandled URL in mock: http://localhost:1317/cosmos/bank/v1beta1/supply
Log: FAILURE chain-osmosis-1 (rpc): Unhandled URL in mock: http://localhost:26657/status
Log: FAILURE chain-osmosis-1 (faucet): Unhandled URL in mock: http://localhost:8007/status
Log: SKIPPED chain-osmosis-1 (exposer): Port not found
Log: FAILURE chain-cosmos-2 (rest): Unhandled URL in mock: http://localhost:1313/cosmos/bank/v1beta1/supply
Log: FAILURE chain-cosmos-2 (rpc): Unhandled URL in mock: http://localhost:26653/status
Log: FAILURE chain-cosmos-2 (faucet): Unhandled URL in mock: http://localhost:8003/status
Log: SKIPPED chain-cosmos-2 (exposer): Port not found
Log: SKIPPED relayer-osmos-cosmos (rest): Port not found
Log: SKIPPED relayer-osmos-cosmos (exposer): Port not found
Log: FAILURE registry (rest): Unhandled URL in mock: http://localhost:8081/chains
Log⬇

Some services failed verification. Please check the logs above.
EndLog⬆"
`;

exports[`StarshipClient verify should handle explorer verification failure 2`] = `""`;

exports[`StarshipClient verify should handle registry verification failure 1`] = `
"Call: setConfig({"name":"osmojs","version":"v0.2.3","chains":[{"id":"osmosis-1","name":"osmosis","image":"pyramation/osmosis:v16.1.0","numValidators":1,"ports":{"rest":1317,"rpc":26657,"faucet":8007}},{"id":"cosmos-2","name":"cosmos","numValidators":1,"ports":{"rest":1313,"rpc":26653,"faucet":8003}}],"relayers":[{"name":"osmos-cosmos","type":"hermes","replicas":1,"chains":["osmosis-1","cosmos-2"]}],"registry":{"enabled":true,"ports":{"rest":8081,"grpc":9091}}})
Log: config again: [object Object]
Call: verify()
Log: Verifying services...
Log: FAILURE chain-osmosis-1 (rest): Unhandled URL in mock: http://localhost:1317/cosmos/bank/v1beta1/supply
Log: FAILURE chain-osmosis-1 (rpc): Unhandled URL in mock: http://localhost:26657/status
Log: FAILURE chain-osmosis-1 (faucet): Unhandled URL in mock: http://localhost:8007/status
Log: SKIPPED chain-osmosis-1 (exposer): Port not found
Log: FAILURE chain-cosmos-2 (rest): Unhandled URL in mock: http://localhost:1313/cosmos/bank/v1beta1/supply
Log: FAILURE chain-cosmos-2 (rpc): Unhandled URL in mock: http://localhost:26653/status
Log: FAILURE chain-cosmos-2 (faucet): Unhandled URL in mock: http://localhost:8003/status
Log: SKIPPED chain-cosmos-2 (exposer): Port not found
Log: SKIPPED relayer-osmos-cosmos (rest): Port not found
Log: SKIPPED relayer-osmos-cosmos (exposer): Port not found
Log: FAILURE registry (rest): Registry not available
Log⬇

Some services failed verification. Please check the logs above.
EndLog⬆"
`;

exports[`StarshipClient verify should handle registry verification failure 2`] = `""`;

exports[`StarshipClient verify should handle relayer verification failure 1`] = `
"Call: setConfig({"name":"osmojs","version":"v0.2.3","chains":[{"id":"osmosis-1","name":"osmosis","image":"pyramation/osmosis:v16.1.0","numValidators":1,"ports":{"rest":1317,"rpc":26657,"faucet":8007}},{"id":"cosmos-2","name":"cosmos","numValidators":1,"ports":{"rest":1313,"rpc":26653,"faucet":8003}}],"relayers":[{"name":"osmos-cosmos","type":"hermes","replicas":1,"chains":["osmosis-1","cosmos-2"]}],"registry":{"enabled":true,"ports":{"rest":8081,"grpc":9091}}})
Log: config again: [object Object]
Call: verify()
Log: Verifying services...
Log: FAILURE chain-osmosis-1 (rest): Unhandled URL in mock: http://localhost:1317/cosmos/bank/v1beta1/supply
Log: FAILURE chain-osmosis-1 (rpc): Unhandled URL in mock: http://localhost:26657/status
Log: FAILURE chain-osmosis-1 (faucet): Unhandled URL in mock: http://localhost:8007/status
Log: SKIPPED chain-osmosis-1 (exposer): Port not found
Log: FAILURE chain-cosmos-2 (rest): Unhandled URL in mock: http://localhost:1313/cosmos/bank/v1beta1/supply
Log: FAILURE chain-cosmos-2 (rpc): Unhandled URL in mock: http://localhost:26653/status
Log: FAILURE chain-cosmos-2 (faucet): Unhandled URL in mock: http://localhost:8003/status
Log: SKIPPED chain-cosmos-2 (exposer): Port not found
Log: SKIPPED relayer-osmos-cosmos (rest): Port not found
Log: SKIPPED relayer-osmos-cosmos (exposer): Port not found
Log: FAILURE registry (rest): Unhandled URL in mock: http://localhost:8081/chains
Log⬇

Some services failed verification. Please check the logs above.
EndLog⬆"
`;

exports[`StarshipClient verify should handle relayer verification failure 2`] = `""`;

exports[`StarshipClient verify should skip disabled services 1`] = `
"Call: setConfig({"name":"osmojs","version":"v0.2.3","chains":[{"id":"osmosis-1","name":"osmosis","image":"pyramation/osmosis:v16.1.0","numValidators":1,"ports":{"rest":1317,"rpc":26657,"faucet":8007}},{"id":"cosmos-2","name":"cosmos","numValidators":1,"ports":{"rest":1313,"rpc":26653,"faucet":8003}}],"relayers":[{"name":"osmos-cosmos","type":"hermes","replicas":1,"chains":["osmosis-1","cosmos-2"]}],"registry":{"enabled":false,"ports":{"rest":8081,"grpc":9091}},"explorer":{"enabled":false}})
Log: config again: [object Object]
Call: verify()
Log: Verifying services...
Log: SUCCESS chain-osmosis-1 (rest): Chain supply is greater than 0
Log: SUCCESS chain-osmosis-1 (rpc): Chain is synced
Log: SUCCESS chain-osmosis-1 (faucet): Chain faucet is working
Log: SKIPPED chain-osmosis-1 (exposer): Port not found
Log: SUCCESS chain-cosmos-2 (rest): Chain supply is greater than 0
Log: SUCCESS chain-cosmos-2 (rpc): Chain is synced
Log: SUCCESS chain-cosmos-2 (faucet): Chain faucet is working
Log: SKIPPED chain-cosmos-2 (exposer): Port not found
Log: SKIPPED relayer-osmos-cosmos (rest): Port not found
Log: SKIPPED relayer-osmos-cosmos (exposer): Port not found
Log⬇

All services verified successfully!
EndLog⬆"
`;

exports[`StarshipClient verify should skip disabled services 2`] = `""`;

exports[`StarshipClient verify should verify all services successfully 1`] = `
"Call: setConfig({"name":"starship","version":"v0.2.3","chains":[{"id":"osmosis-1","name":"osmosis","image":"pyramation/osmosis:v16.1.0","numValidators":1,"ports":{"rest":1317,"rpc":26657,"faucet":8007}},{"id":"cosmos-2","name":"cosmos","numValidators":1,"ports":{"rest":1313,"rpc":26653,"faucet":8003}}],"relayers":[{"name":"osmos-cosmos","type":"hermes","replicas":1,"chains":["osmosis-1","cosmos-2"]}],"registry":{"enabled":true,"ports":{"rest":8081,"grpc":9091}}})
Log: config again: [object Object]
Call: verify()
Log: Verifying services...
Log: SUCCESS chain-osmosis-1 (rest): Chain supply is greater than 0
Log: SUCCESS chain-osmosis-1 (rpc): Chain is synced
Log: SUCCESS chain-osmosis-1 (faucet): Chain faucet is working
Log: SKIPPED chain-osmosis-1 (exposer): Port not found
Log: SUCCESS chain-cosmos-2 (rest): Chain supply is greater than 0
Log: SUCCESS chain-cosmos-2 (rpc): Chain is synced
Log: SUCCESS chain-cosmos-2 (faucet): Chain faucet is working
Log: SKIPPED chain-cosmos-2 (exposer): Port not found
Log: SKIPPED relayer-osmos-cosmos (rest): Port not found
Log: SKIPPED relayer-osmos-cosmos (exposer): Port not found
Log: SUCCESS registry (rest): Registry is working
Log⬇

All services verified successfully!
EndLog⬆"
`;

exports[`StarshipClient verify should verify all services successfully 2`] = `""`;
